### ISSUE TITLE
Error when fetching from remote source (Fixed on 2.7 and above)

### ISSUE THREAD

**Issue by AarronTame** [michelecocuccio/simple-react-lightbox#28]
Hi, I've implemented the code a per the documentation. When using fetch from remote source I get a `Uncaught TypeError: Cannot read property 'nodeName' of undefined` error.

I'm using a `useEffect` and `fetch` to get the data and setting that to local state, it appears to be trying to get the images before the component has finished rendering.

If i statically place images on the page, the functionality works correctly.

Any help would be greatly appreciated.

```
Set up
nextJs: 9.3.1
SRL: 2.6.8
```

**Comment by michelecocuccio** [comment]
@AarronTame hi thanks for the heads up.
You can wrap the ```<SRLWrapper />``` without having images and it won't throw an error. It will simply do nothing and the context with the images will be just an empty array. I think yours is an implementation issue. If you are fetching data from an external API you will probably get a promise. Maybe you can just wait the problem to be resolved and then return the images inside the ```<SRLWrapper>``` or even better  you render a "Loading..." Component and when the promise is resolved you can render the Component with ```<SRLWrapper>```. I will do some testing anyway with data API this weekend as I am implementing a feature and I will see if I get the same issue. 
If you want to put your code on a CodeSandbox I could try to check what's going on in your case.

**Comment by AarronTame** [comment]
@michelecocuccio Hi, thanks for the quick response.
You are right, it looks like part of it is due to implementation, but i also believe there is a bug.

I have two scenarios for implementation:

**Scenario 1:** _`useEffect` to get initial data set, then the scroller handles the additional requests_
I am using a infinite scroller (`react-infinite-scroll-component`) to fetch the images, it correctly id's the images with `elementX` on the initial get and appear in the context, but when I then fetch subsequent images, they appear in the context, but the html elements do not get the `id` attribute.

![image](https://user-images.githubusercontent.com/27616510/79566079-d6375400-80a9-11ea-9511-c3769440b42f.png)

`id="element9"` is the last image from the initial get, after this the images are from additional requests which don't have the `id` property.

![image](https://user-images.githubusercontent.com/27616510/79566128-ee0ed800-80a9-11ea-91c9-30378f6111e9.png)

**Scenario 2:** _This component handles the get request when needed, so no `useEffect`_
When using an additional gallery component (in this case `egjs  InfiniteGrid`), the context show the gallery component and not the children images, and again doesn't add the `id` property to the images.

![image](https://user-images.githubusercontent.com/27616510/79564971-97080380-80a7-11ea-83f8-79f41a4b1191.png)

``` jsx
<SRLWrapper>
     <JustifiedLayout>
         {images.map(image => <Image />)}
     </JustifiedLayout
</SRLWrapper
```
If you need anymore information let me know.

**Comment by michelecocuccio** [comment]
Thanks for the detailed explanation. Ok I can see the issue there. 
You are adding ADDITIONAL images. Unfortunately unless you re-render your component the ```<SRLWrapper>``` won't be able to find additional images. I never tried to add ADDITIONAL images but I can experiment a little bit when I have some time and probably add a fix for this scenarios. Keep in mind that this is not a "Bug" per se, is more like an additional implementation. 
I think it will work if you re-render the component but I guess that giving your App you don't want that, you are just fetching images on scroll. 

By the way are you using useState to keep track of your images? so like doing 
```const [images, setImages] = useState([])```
and then when you fetch more images (when you scroll from my understanding) you use setImages to update the array. That should re-render the component and the ```SRLWrapper``` should re-render. 

Sorry is a bit difficult without looking at the code, I am just guessing there.

**Comment by michelecocuccio** [comment]
@AarronTame I am sorry you were right. I tested it recently and in some cases it was giving me the ``Uncaught TypeError: Cannot read property 'nodeName' of undefined error.`` and in some instances it wasn't. Even when it wasn't, unfortunately because no props were passed or any state wasn't changed the component wasn't re-rendering as well. So I implemented this in version 2.7. You can check the example on the demo. I am not using your implementation on loading while scrolling, but it should re-render correctly. Please feel free to try and let me know. I cannot test all the implementations but if it works with fetch it should work with axios as well if you implement the logic correctly.

**Comment by AarronTame** [comment]
@michelecocuccio Thanks for the update, it now works as expected. Thanks for a great solution!!!
